README.md

```diff
@@ -1,115 +1,624 @@
-sys = ['The dog bit the man.', "It wasn't surprising.", 'The man had just bitten him.']
-bleu = sacrebleu.corpus_bleu(sys, refs)
-print(bleu.score)
-```
-
-# MOTIVATION
-
-Comparing BLEU scores is harder than it should be.
-Every decoder has its own implementation, often borrowed from Moses, but maybe with subtle changes.
-Moses itself has a number of implementations as standalone scripts, with little indication of how they differ (note: they mostly don't, but `multi-bleu.pl` expects tokenized input).
-Different flags passed to each of these scripts can produce wide swings in the final score.
-All of these may handle tokenization in different ways.
-On top of this, downloading and managing test sets is a moderate annoyance.
-Sacre bleu!
-What a mess.
-
-SacreBLEU aims to solve these problems by wrapping the original Papineni reference implementation together with other useful features.
-The defaults are set the way that BLEU should be computed, and furthermore, the script outputs a short version string that allows others to know exactly what you did.
-As an added bonus, it automatically downloads and manages test sets for you, so that you can simply tell it to score against 'wmt14', without having to hunt down a path on your local file system.
-It is all designed to take BLEU a little more seriously.
-After all, even with all its problems, BLEU is the default and---admit it---well-loved metric of our entire research community.
-Sacre BLEU.
+In [1]: from sacrebleu.metrics import BLEU, CHRF, TER
+   ...: 
+   ...: refs = [ # First set of references
+   ...:          ['The dog bit the man.', 'It was not unexpected.', 'The man bit him first.'],
+   ...:          # Second set of references
+   ...:          ['The dog had bit the man.', 'No one was surprised.', 'The man had bitten the dog.'],
+   ...:        ]
+   ...: sys = ['The dog bit the man.', "It wasn't surprising.", 'The man had just bitten him.']
+
+In [2]: bleu = BLEU()
 
-# LICENSE
+In [3]: bleu.corpus_score(sys, refs)
+Out[3]: BLEU = 48.53 82.4/50.0/45.5/37.5 (BP = 0.943 ratio = 0.944 hyp_len = 17 ref_len = 18)
 
-SacreBLEU is licensed under the Apache 2.0 License.
+In [4]: bleu.get_signature()
+Out[4]: nrefs:2|case:mixed|eff:no|tok:13a|smooth:exp|version:2.0.0
+
+In [5]: chrf = CHRF()
+
+In [6]: chrf.corpus_score(sys, refs)
+Out[6]: chrF2 = 59.73
+```
 
-# CREDITS
+### Variable Number of References
+
+Let's now remove the first reference sentence for the first system sentence `The dog bit the man.` by replacing it with either `None` or the empty string `''`.
+This allows using a variable number of reference segments per hypothesis. Observe how the signature changes from `nrefs:2` to `nrefs:var`:
+
+```python
+In [1]: from sacrebleu.metrics import BLEU, CHRF, TER
+   ...: 
+   ...: refs = [ # First set of references
+                 # 1st sentence does not have a ref here
+   ...:          ['', 'It was not unexpected.', 'The man bit him first.'],
```

Comment:
So the first sentence has no reference at all?
Such example fails with RuntimeError: Empty or `None` reference sentence found.
I think the use case we want to show here is that the first sentences has only a single reference (and is missing the second reference), i.e.
`['The dog bit the man.', 'It was not unexpected.', 'The man bit him first.'],`

Reply:
No first sentence has a reference but coming from the second set of references. The order shouldn't matter, I don't know why I removed one of the refs from the first list instead of the second. I'll see this.

Reply:
Currently, it is missing in both sets of references.
I agree the order of alternative references shouldn't matter, so I am OK with showing a missing sentence in the first set of references.

Reply:
ahh, sorry that's indeed an error.

Reply:
fixed.